spyder/utils/programs.py

```diff
@@ -967,6 +968,19 @@ def check_python_help(filename):
         return False
 
 
+def get_python_site_packages(pyexec):
+    """Get the site-packages for a given python executable"""
+    script = 'import site; print(site.getsitepackages())'
+    try:
+        proc = run_program('env', ['-i', pyexec, '-c', script])

```

Comment:
Same here as with others PRs, spyder is a cross platform program so things should also work on windows, or we need to provide a different way so it works on windows.

I know this is work for a mac standalone app, but it seems to be using code that will only work on mac and in the process it seems it will break spyder on windows.

Reply:
Thanks for looking at this. Yes, I'll put in some flow control, as I mentioned in #12235. 

Tangentially, while my primary motivation is for the Mac app, if we found a suitable equivalent to `env -i` on windows, then we could potentially support launching Spyder from non-conda environments, independent of platform.

Reply:
We can have @dalthviz (our win expert on the team) take a look at this but in the meantime if you need to change anything to make the MacApp work, I would say we should always use control flow and even (as mentioned in the other PR) consider adding a constant that will be True only if using the MacApp

Reply:
> consider adding a constant that will be True only if using the MacApp

There's one already: `running_in_mac_app` (in `spyder/config/base.py`).

Reply:
> There's one already: `running_in_mac_app` (in `spyder/config/base.py`).

That's good to know! However, I think the benefit is not limited to the Mac app so I think the flow control will be best.